### DISCUSSION TITLE
Server side fetch with Fetch API results to Auth session missing

### DISCUSSION THREAD

**Discussion by bennik88** [supabase/supabase#27552] [Questions]
**Problem** 
When I make a request client site with the Fetch API I'm getting the user in my route.tsx file with `supabase.auth.getUser();` every time but when I'm making a request from a server component with the Fetch API my Auth session is missing in my route.tsx. And I can't figure out why.
When I use `supabase.auth.getUser();` in the server component directly the user is present and I'm able to make calls to the database directly using the supbabase client `const supabase = createClient();` in the server component. What is the problem do I need to set the headers manually in Server components for the fetch request, but then it's working for the client components? I'm using the new supabase SSR see my packages below.

**What I've tried**
As the middleware with `updateSession` already should have set the cookies and I confirmed that by logging the user is there on the server before the fetchUserData, I tried to use `credentials: "same-origin"`, and `"include"` with my `fetch()` call. But it still didn't work.

**Code:** 
page.tsx
```js
"use server"

export default async function CheckoutSuccessPage() {
  const supabase = createClient();
  const {
    data: { user }, // This works on a server component
  } = await supabase.auth.getUser(); 

  async function fetchUserData() { // On client side this works, on server side it doesn't
    try {
      const res = await fetch(process.env.NEXT_PUBLIC_APP_URL + GET_USER_DATA, {
        method: "GET",
      });
      if (!res.ok) throw new Error(res.statusText);

      const data = await res.json();
      return data;
    } catch (error: any) {
      console.log({ error });
      console.error(error.message);
      throw new Error(error?.message);
    }
  }
```

route.tsx
```js
import { createClient } from "@/utils/supabase/server";

export async function GET(req: Request) {
  if (!process.env.NEXT_PUBLIC_SUPABASE_URL || !process.env.SUPABASE_ROLE_KEY) {
    throw new Error("Environment is not set.");
  }

  const supabase = createClient();

  const {
    data: { user }, // Here my user is missing and I get an Auth session missing error when I do the fetch from my server component
    error,
  } = await supabase.auth.getUser();

  if (!user || error)
    return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
```

**Stack:**
```
"@supabase/ssr": "^0.3.0"
"@supabase/supabase-js": "^2.43.4"
"supabase": "^1.176.10"
"react": "^18.3.1"
```

**Comment by saffiullahkhugyani** [comment]
I am stuck in the same boat

**Comment by ElectricCodeGuy** [comment]
https://github.com/ElectricCodeGuy/SupabaseAuthWithSSR
I made a simple code example here you can try

**Comment by pstachula-dev** [comment]
I checked my code and it works fine in API routes.

Please check this tutorial https://supabase.com/docs/guides/auth/server-side/nextjs?queryGroups=router&router=app
It looks like you are missing cookies in createServerClient - it should be async, in your example you are call `createClient` without await.

**Comment by MrVibe** [comment]
I think you most likely need to pass the cookie functions in your createClient on server to identify the correct supabase instance instead of creating a new one.
